### DISCUSSION TITLE
New power limit

### DISCUSSION THREAD

**Discussion by seerge** [seerge/g-helper#481] [General]
Hello, I have added one more power limit slider to the app. On G14 2022 it's responsible for what HWInfo calls "APU STAPM" limit 

I suspect that on other models this can be a bit different. 

Good way to test - is:

1. Run [HWInfo](https://www.hwinfo.com/download/)
2. Set some value for APU slider (and click Apply of course), and then
3. Run [Cinebench23](https://www.maxon.net/en/downloads/cinebench-r23-downloads) or any similar benchmark
4. See what value seem to be limited 

![Screenshot 2023-05-29 222604](https://github.com/seerge/g-helper/assets/5920850/15c49e17-256a-4c58-b372-e3b2b7202cd8)

**Comment by Inaiso** [comment]
Just a quick question, so this will limit the power of the APU which is the AMD Radeon 600M on the 2022 model? Or will it do more than that?

**Reply by seerge** [reply]
For me  APU STAPM limit in HWInfo seem to be a total limit over everything (i.e. CPU + iGPU + dGPU), so it's very similar to platform. In Eco mode - it's obviously only APU (as there is no GPU)

**Reply by NJinsei** [reply]
I used it but it doesn't seem to limit the DGPU like you said as lowering or increasing didn't seem to affect the DGPU's overall power draw

**Reply by seerge** [reply]
@NJinsei on G14 2022? It affects total power draw, so it's pretty much identical to what platform slider does

**Reply by NJinsei** [reply]
No, it's on the Flow X16 2022 all the sliders solely only affect the CPU.

**Reply by seerge** [reply]
@NJinsei ok, good to know. But what the difference between the one called "APU" and "CPU" in your case?

**Reply by seerge** [reply]
@NJinsei can you set some non-standard values to all 3 sliders (like 103, 83, 73 or so, so they definetely don't match anything default) and then download and run https://github.com/FlyGoat/RyzenAdj/releases/download/v0.13.0/ryzenadj-win64.zip in powershell as admin

```
./ryzenadj -i
```
It will output all currently set values , and then you can match what slider sets what value 

To illustrate what i mean : 

![Screenshot 2023-05-30 172314](https://github.com/seerge/g-helper/assets/5920850/7b1a8c20-629a-41ea-813a-7e4b62e3a0c5)

Then I can label sliders properly at least ... :)

**Reply by NJinsei** [reply]
Both sliders affect the CPU power, but the APU STAPM overrides the CPU power limit. However, unlike in your screenshot, the APU STAPM does not match the limits and it stays around 10-14 W. So I'm not sure what it's actually doing.

**Reply by NJinsei** [reply]
> @NJinsei can you set some non-standard values to all 3 sliders (like 103, 83, 73 or so, so they definetely don't match anything default) and then download and run https://github.com/FlyGoat/RyzenAdj/releases/download/v0.13.0/ryzenadj-win64.zip in powershell as admin
> 
> ```
> ./ryzenadj -i
> ```
> 
> It will output all currently set values , and then you can match what slider sets what value
> 
> To illustrate what i mean :
> 
> ![Screenshot 2023-05-30 172314](https://user-images.githubusercontent.com/5920850/242015950-7b1a8c20-629a-41ea-813a-7e4b62e3a0c5.png)
> 
> Then I can label sliders properly at least ... :)

Alright

**Reply by NJinsei** [reply]
<img width="1132" alt="image" src="https://github.com/seerge/g-helper/assets/89779681/90bbd894-1abb-41cf-8fff-966f2ae0baae">
This is what I got, I tried reapplying it and everything but only the APU STAPM seems to apply.

**Reply by seerge** [reply]
@NJinsei good, do you mind to set Turbo as a base and then set APU to something higher than platform ?

For example : Platform 83, CPU : 73, APU : 93 

(i suspect "cpu" slider does nothing for you, just want to make sure what platform does)

**Reply by NJinsei** [reply]
<img width="1128" alt="image" src="https://github.com/seerge/g-helper/assets/89779681/dbb35d59-ed9c-464b-8143-6e1781fd4b77">
Yeah it seems your correct CPU doesn't seem to affect it.

**Reply by seerge** [reply]
@NJinsei ok, great. So platform = CPU slow limit (i.e. for sustained loads), APU = CPU fast limit i.e. for short boosts (usually it's 2 mins max)

**Reply by seerge** [reply]
@NJinsei  Check this build, it should show only 2 sliders for you (and labeled correctly). and thanks for testing
[GHelper.zip](https://github.com/seerge/g-helper/files/11602019/GHelper.zip)

**Reply by NJinsei** [reply]
Ah alright that makes sense then

**Reply by NJinsei** [reply]
Thank you for the updated build, although the Miniled toggle seems to be labelled incorrectly, also the CPU slider is still visible.
<img width="69" alt="image" src="https://github.com/seerge/g-helper/assets/89779681/4bab5eca-38b3-44da-9c80-dc9a796983ca">

**Reply by seerge** [reply]
@NJinsei right, i was adjusting some translations and mixed labels :) 

This one should have correct label [GHelper.zip](https://github.com/seerge/g-helper/files/11602272/GHelper.zip)

Btw, since you are here, do you mind to try if Eco / Standard gpu switching work ok in this build ? I'm testing a feature when app will stop all nvidia processes prior to disabling GPU (it can sometimes help with situations if GPU refuses to get disabled)

**Reply by NJinsei** [reply]
The feature didn't seem to work for me. After switching it on it failed to close any programs running on the DGPU and did not switch back to IGPU mode either. Even after turning off the feature, I could not switch off the DGPU.

**Reply by seerge** [reply]
@NJinsei that's strange.  Can you try this build? But mb run it as admin (i suspect some apps could require admin rights to get closed)

And PPT labels should be correct for your case here (i.e. only 2 sliders  called CPU slow and CPU fast)

[GHelper.zip](https://github.com/seerge/g-helper/files/11602377/GHelper.zip)

**Reply by NJinsei** [reply]
Just tried the feature and it seems to work without any issues.

**Reply by NJinsei** [reply]
Also the title which shows the current power limits of your performance mode doesn't match the new PPT limits that are now applied.
<img width="826" alt="image" src="https://github.com/seerge/g-helper/assets/89779681/11c369c6-e47f-49c5-a175-6506885b93d1">

**Reply by seerge** [reply]
@NJinsei good eye, this build will show Fast limit in title instead
[GHelper.zip](https://github.com/seerge/g-helper/files/11603582/GHelper.zip)

As for OSD and hotkeys, honestly i didn't change that anywhere in last releases. Keep in mind that if optimization service is not running - g-helper has to be running (in tray) to keep hotkeys working

Also can you post a log from GPU switching (in both directions) ? i want to see if it logs everything it closes correctly

**Reply by NJinsei** [reply]
Alright done here are the logs, everything seems to be working now. Also another thing I've noticed in general is that the app doesn't seem to close the actual NVIDIA Settings and services after it launches them once switched backed into eco mode.
[log.txt](https://github.com/seerge/g-helper/files/11604513/log.txt)

**Reply by seerge** [reply]
@NJinsei thanks, how is that process called ? i can add it to the kill list :)

This is what g-helper has killed so far

```
30/05/2023 20:23:36: Stopped: NVDisplay.Container
30/05/2023 20:23:36: Stopped: NVDisplay.Container
30/05/2023 20:23:36: Stopped: nvcontainer
30/05/2023 20:23:36: Stopped: nvcontainer
30/05/2023 20:23:36: Stopped: nvcontainer
30/05/2023 20:23:36: Stopped: Stray-Win64-Shipping
```

**Reply by NJinsei** [reply]
Here are the NVIDIA Services before it's switched, although I'm not sure why they're there. 
<img width="640" alt="Screenshot 2023-05-30 204135" src="https://github.com/seerge/g-helper/assets/89779681/0c530957-41d4-4706-b6fc-86d92332a4ce">

Here are the ones after.
<img width="626" alt="Screenshot 2023-05-30 204317" src="https://github.com/seerge/g-helper/assets/89779681/f3e5f12b-bf3b-4b67-b62a-749a57612e35">

It seems the ones that it leaves behind are NVIDIA Shadow Play and NVIDIA Share.

**Reply by seerge** [reply]
@NJinsei that's a lot of nvidia stuff :) i can imagine it restarts some of own things after it's killed, but main thing - it doesn't block GPU disabling.  I don't know which process is nvidia settings there, so I will leave current kill list as (i don't have nvidia device by myself atm, so can't test properly)

**Reply by NJinsei** [reply]
That's alright I understand. Thank you for your hard work on this app, as it's become irreplaceable on my ASUS device.
